### DISCUSSION TITLE
MS AppX / MSIX packager

### DISCUSSION THREAD

**Discussion by djgilcrease** [goreleaser/nfpm#345] [Ideas]
I am looking for help converting the MS MSIX packager and unpackager to golang so that we can use it in NFPM
https://github.com/microsoft/msix-packaging/blob/master/src/msix/

Are there any MSI experts out there reading this that would be willing to help?

**Comment by caarlos0** [comment]
There are some people interested in https://github.com/goreleaser/goreleaser/issues/1295 

Will ping them there :)

**Comment by erikgeiser** [comment]
This is the first time I'm hearing about MSIX. I think supporting this would be great for `goreleaser`. However, my concern is that a lot of concepts `nfpm` won't translate that well. We only have some rather niche per-packager config fields because in general `apt`, `rpm` and `apk` are so similar compared to anything I can imagine for Windows. Also I can't imagine a `nfpm` config for both Linux and Windows targets, primarily due to the filesystem layout.

Therefore, it may be a better idea to start an `nfpm` spin-off only for Windows under the umbrella of `goreleaser`.

**Reply by djgilcrease** [reply]
Maybe, but in looking though the code much of the oddities such as guids and such are all actually generated as uuid5s from fields we already have.

I am first concerned with creating a library that can build a msix package that will install properly, then I will work on integrating it into nfpm.

**Comment by abemedia** [comment]
Microsoft has an SDK for that, however it would require using CGO: https://github.com/Microsoft/msix-packaging
